### ISSUE TITLE
ProtectSystem=strict disables ProtectHome=tmpfs

### ISSUE THREAD

**Issue by embe** [systemd/systemd#11276]
**systemd version the issue has been seen with**
240

**Used distribution**
Gentoo

**Expected behaviour you didn't see**
Unit with `ProtectSystem=strict` and `ProtectHome=tmpfs` should have a tmpfs mounted at `/home`.

**Unexpected behaviour you saw**
The unit has access to normal `/home`.

**Steps to reproduce the problem**
```
# systemd-run --pipe --property=ProtectHome=tmpfs stat -fc '%T' /home 
Running as unit: run-u286.service
tmpfs
# systemd-run --pipe --property=ProtectHome=tmpfs --property=ProtectSystem=strict stat -fc '%T' /home
Running as unit: run-u287.service
ext2/ext3
```

**Comment by yuwata** [comment]
Hmm, when I implemented this, I got the comment https://github.com/systemd/systemd/pull/7908#discussion_r167519463...

**Comment by embe** [comment]
> Hmm, when I implemented this, I got the comment [#7908 (comment)](https://github.com/systemd/systemd/pull/7908#discussion_r167519463)...

Regarding that comment: if I understand the code, `ProtectHome=read-only` + `ReadWriteDirectory=/root` doesn't make `/root` read-write, so the current tmpfs behaviour is not consistent. I think tmpfs should be much higher on the priority list.
@poettering?

**Comment by yuwata** [comment]
@embe I've created #11283. If possible, please test it. Thanks.
